trieste/acquisition/optimizer.py

```diff
@@ -35,24 +34,83 @@
 AcquisitionOptimizer = Callable[[SP, AcquisitionFunction], TensorType]
 """
 Type alias for a function that returns the single point that maximizes an acquisition function over
-a search space. For a search space with points of shape S, and acquisition function with input shape
-[...] + S output shape [..., 1], the :data:`AcquisitionOptimizer` return shape should be [1] + S.
+a search space. For a search space with points of shape [D], and acquisition function with input
+shape [..., B, D] output shape [..., 1], the :const:`AcquisitionOptimizer` return shape should be
+[B, D].
 """
 
 
-@singledispatch
-def optimize(space: Box | DiscreteSearchSpace, target_func: AcquisitionFunction) -> TensorType:
+def automatic_optimizer_selector(
+    space: DiscreteSearchSpace, target_func: AcquisitionFunction
+) -> TensorType:
     """
+    A wrapper around our :const:`AcquisitionOptimizer`s. This class performs
+    an :const:`AcquisitionOptimizer` appropriate for the
+    problem's :class:`~trieste.space.SearchSpace`.
+
     :param space: The space of points over which to search, for points with shape [D].
-    :param target_func: The function to maximise, with input shape [..., D] and output shape
+    :param target_func: The function to maximise, with input shape [..., 1, D] and output shape
         [..., 1].
-    :return: The point in ``space`` that maximises ``target_func``, with shape [1, D].
+    :return: The batch of points in ``space`` that maximises ``target_func``, with shape [1, D].
+    """
+
+    if isinstance(space, Box):
+        optimizer = optimize_continuous
+    elif isinstance(space, DiscreteSearchSpace):
+        optimizer = optimize_discrete
+    else:
+        raise NotImplementedError(
+            """ No optimizer currentely supports acquisition function
+            maximisation over this search space"""
+        )
+
+    return optimizer(space, target_func)
+
+
+def batchify(
+    batch_size_one_optimizer: AcquisitionOptimizer,
```

Comment:
```suggestion
    batch_size_one_optimizer: AcquisitionOptimizer[SP],
```

Reply:
What does this do?

Reply:
say I have `optimize_continuous` which is an `AcquisitionOptimizer[Box]`, i.e. it's an optimizer over `Box` search spaces. If I call `new_optimizer = batchify(optimize_continuous, 3)`, having the same type variable on both the input and output means that I can be sure that `new_optimizer` is also an optimizer over `Box`.